### ISSUE TITLE
Validation error message in textbox not show if validation falls at the beginning

### ISSUE THREAD

**Issue by nejcgalof** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2182]
If we look in demo fields, more specific "NameTextBox", error message is not shown, but line is red.
I do some tests and it shown when valid status change (from false at the beginning to true and then to false again).
If we create validation rule, where we check if length of text in textbox is bigger then 2, error not shown even if we type first letter.

Expected: If the line (textbox) is red, then the error message must also be displayed. If validation on first time is not correct, then display error message.

**Comment by KHALED-LAKEHAL** [comment]
```
<Window x:Class="StoreManagement.Windows.Login"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:helpers="clr-namespace:StoreManagement.Utilities.Helpers"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"   
    xmlns:behaviors="clr-namespace:StoreManagement.Utilities.Behaviors"
    xmlns:i="http://schemas.microsoft.com/expression/2010/interactivity"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:validations="clr-namespace:StoreManagement.Utilities.Validations"
    mc:Ignorable="d" Width="370" Height="580" DataContext="{Binding RelativeSource={RelativeSource Self}}">
    <TextBox x:Name="TextBoxControl">
        <i:Interaction.Triggers>
            <i:EventTrigger EventName="LostFocus">
                <i:InvokeCommandAction Command="{Binding OnLostFocusCommand, Source={x:Static helpers:ValidationHelper.Instance}}" CommandParameter="{Binding ElementName=TextBoxControl}" />
            </i:EventTrigger>
        </i:Interaction.Triggers>
        <TextBox.Text>
            <Binding Path="Text" UpdateSourceTrigger="PropertyChanged" Mode="TwoWay">
                <Binding.ValidationRules>
                    <!-- Validation Rules -->
                </Binding.ValidationRules>
            </Binding>
        </TextBox.Text>
    </TextBox>
</Window>



using System;
using System.Windows;
using System.Windows.Data;
using System.Windows.Input;
using System.Windows.Controls;
using StoreManagement.Controls;


public class ValidationHelper
{
    public ICommand OnLostFocusCommand { get; set; }

    private static ValidationHelper _instance;
    public static ValidationHelper Instance
    {
        get
        {
            if (_instance == null)
            {
                _instance = new ValidationHelper();
            }

            return _instance;
        }
    }

    public ValidationHelper()
    {
        OnLostFocusCommand = new RelayCommand(OnLostFocusCommandExecute, OnLostFocusCommandCanExecute);
    }

    private void OnLostFocusCommandExecute(Object arg)
    {
        OnLostFocus(arg, new RoutedEventArgs());
    }

    private Boolean OnLostFocusCommandCanExecute(Object arg)
    {
        return true;
    }

    private void OnLostFocus(Object sender, RoutedEventArgs e)
    {
        CheckBindingExpression(sender, false);
    }

    private static void CheckBindingExpression(Object control, Boolean IsClear)
    {
        switch (control.GetType().ToString())
        {
            case "System.Windows.Controls.TextBox":
                SetBindingExpression(((TextBox)control).GetBindingExpression(TextBox.TextProperty), IsClear);
                break;
            case "System.Windows.Controls.PasswordBox":
                SetBindingExpression(((PasswordBox)control).GetBindingExpression(PasswordHelper.PasswordProperty), IsClear);
                break;
        }
    }

    private static void SetBindingExpression(BindingExpression bindingExpression, Boolean IsClear)
    {
        if (bindingExpression != null)
        {
            if (IsClear)
            {
                Validation.ClearInvalid(bindingExpression);
            }
            else
            {
                bindingExpression.UpdateSource();
            }
        }
    }
}
```

**Comment by nejcgalof** [comment]
I fix it with PR #2197

**Comment by Keboo** [comment]
Closed by #2197
